### ISSUE TITLE
Sync all the things

### ISSUE THREAD

**Issue by JosuaKrause** [remram44/gitobox#15]
Might be a cool feature --> having an API for writing modules and converting the current system to those modules.
Then gitobox's job becomes that of a daemon that sync's module A (eg. watchdog) with module B (eg. git). (fully orthogonal ie. could also sync two different places in the file system)

API:
- get notified
- fetch data
- commit data

Might be forgetting something...
Just an idea :)

**Comment by remram44** [comment]
Yes, there is no real reason to only sync Dropbox with Git. You could totally sync Dropbox with Google Drive, Dropbox with another Dropbox, or Git with Git (why would you?!)

The only limitation here is that there is currently only one "working directory" -- Git checks out and commits from the Dropbox folder itself, there is no copy. But I'm sure this is no problem if the API is thought out carefully.
